dependencydiff/dependencydiff.go

```diff
@@ -0,0 +1,162 @@
+	ghRepoClient                          clients.RepoClient
+	ossFuzzClient                         clients.RepoClient
+	vulnsClient                           clients.VulnerabilitiesClient
+	ciiClient                             clients.CIIBestPracticesClient
+	changeTypesToCheck                    map[pkg.ChangeType]bool
+	checkNamesToRun                       []string
+	dependencydiffs                       []dependency
+	results                               []pkg.DependencyCheckResult
+}
+
+// GetDependencyDiffResults gets dependency changes between two given code commits BASE and HEAD
+// along with the Scorecard check results of the dependencies, and returns a slice of DependencyCheckResult.
+// TO use this API, an access token must be set following https://github.com/ossf/scorecard#authentication.
+func GetDependencyDiffResults(
+	ctx context.Context, ownerName, repoName, baseSHA, headSHA string, scorecardChecksNames []string,
+	changeTypesToCheck map[pkg.ChangeType]bool) ([]pkg.DependencyCheckResult, error) {
+	// Fetch the raw dependency diffs.
+	dCtx := dependencydiffContext{
+		logger:             log.NewLogger(log.InfoLevel),
+		ownerName:          ownerName,
+		repoName:           repoName,
+		baseSHA:            baseSHA,
+		headSHA:            headSHA,
+		ctx:                ctx,
+		changeTypesToCheck: changeTypesToCheck,
+		checkNamesToRun:    scorecardChecksNames,
+	}
+	err := fetchRawDependencyDiffData(&dCtx)
+	if err != nil {
+		return nil, fmt.Errorf("error in fetchRawDependencyDiffData: %w", err)
+	}
+
+	// Initialize the repo and client(s) corresponding to the checks to run.
+	err = initRepoAndClientByChecks(&dCtx)
+	if err != nil {
+		return nil, fmt.Errorf("error in initRepoAndClientByChecks: %w", err)
+	}
+	getScorecardCheckResults(&dCtx)
+	return dCtx.results, nil
+}
+
+func initRepoAndClientByChecks(dCtx *dependencydiffContext) error {
+	// Create the repo and the corresponding client if its check needs to run.
+	ghRepo, err := githubrepo.MakeGithubRepo(path.Join(dCtx.ownerName, dCtx.repoName))
+	if err != nil {
+		return fmt.Errorf("error creating the github repo: %w", err)
+	}
+	dCtx.ghRepo = ghRepo
+	dCtx.ghRepoClient = githubrepo.CreateGithubRepoClient(dCtx.ctx, dCtx.logger)
+	// Initialize these three clients as nil at first.
+	var ossFuzzClient clients.RepoClient
+	for _, cn := range dCtx.checkNamesToRun {
+		switch cn {
+		case checks.CheckFuzzing:
+			ossFuzzClient, err = githubrepo.CreateOssFuzzRepoClient(dCtx.ctx, dCtx.logger)
+			if err != nil {
+				return fmt.Errorf("error initializing the oss fuzz repo client: %w", err)
+			}
+			dCtx.ossFuzzClient = ossFuzzClient
+		case checks.CheckVulnerabilities:
+			dCtx.vulnsClient = clients.DefaultVulnerabilitiesClient()
+		case checks.CheckCIIBestPractices:
+			dCtx.ciiClient = clients.DefaultCIIBestPracticesClient()
+		}
+	}
+	return nil
+}
+
+func initScorecardChecks(checkNames []string) checker.CheckNameToFnMap {
```

Comment:
Can we call the existing function that already exist in scorecard?

Reply:
found `policy/GetEnabled` that does the same thing, removed mine, thanks!